Sprint 4

LightKrite opened the pull request:
Пулл реквест для 4 спринта обучения в Яндекс.Практикум

LightKrite committed:
-Добавлены шрифты, цвета, прописан код для стандартных цветов приложения.

LightKrite committed:
Добавлен LaunchScreen

LightKrite committed:
- Задан загрузочный экран
- задан основной экран приложения

LightKrite committed:
- Реализована логика приложения.

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,166 @@ import UIKit
 final class MovieQuizViewController: UIViewController {
     // MARK: - Lifecycle
+    
+    @IBAction private func YesButtonClicked(_ sender: Any) {
```
Не забывай, что все названия свойств и функций мы пишем с маленькой буквы

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,166 @@ import UIKit
+        let title: String
+        let text: String
+        let buttonText: String
+    }
```
Эти модели можно вывести за пределы класса) кто знает, может они пригодятся и в другом месте, а тут они закрыты внутри класса

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,166 @@ import UIKit
+    }
+    
+    
+    
```
Лишние пустые строки

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,166 @@ import UIKit
+            message: result.text,
+            preferredStyle: .alert)
+        
+        let action = UIAlertAction(title: result.buttonText, style: .default) { _ in
```
😱  Тут необходимо прописывать [weak self], чтобы избежать цикла сильных ссылок. 
Можно взять за правило - что всегда, при обращении к self. из замыкания, нужно писать [weak self], чтобы код был безопасным)

LightKrite replied:
> 😱 Тут необходимо прописывать [weak self], чтобы избежать цикла сильных ссылок. Можно взять за правило - что всегда, при обращении к self. из замыкания, нужно писать [weak self], чтобы код был безопасным)

я если честно, так и не разобрался, где и как их надо проставить( 
Не могли бы вы подсказать, где про это можно почитать?

LightKrite committed:
-Добавлена блокировка кнопок после их нажатия
-выведен функционал кнопок в отдельные функции для устранения дублей кода.
-небольшие косметические изменения кода
